openml/flows/sklearn_converter.py

```diff
@@ -305,21 +307,36 @@ def _extract_information_from_model(model):
 
         if (isinstance(rval, (list, tuple)) and len(rval) > 0 and
                 isinstance(rval[0], (list, tuple)) and
-                [type(rval[0]) == type(rval[i]) for i in range(len(rval))]):
+                all([isinstance(rval[i], type(rval[0]))
+                     for i in range(len(rval))])):
 
-            # Steps in a pipeline or feature union, or base classifiers in voting classifier
+            # Steps in a pipeline or feature union, or base classifiers in
+            # voting classifier
             parameter_value = list()
             reserved_keywords = set(model.get_params(deep=False).keys())
 
             for sub_component_tuple in rval:
-                identifier, sub_component = sub_component_tuple
+                identifier = sub_component_tuple[0]
+                sub_component = sub_component_tuple[1]
                 sub_component_type = type(sub_component_tuple)
+                if not 2 <= len(sub_component_tuple) <= 3:
+                    # length 2 is for {VotingClassifier.estimators,
+                    # Pipeline.steps, FeatureUnion.transformer_list}
+                    # length 3 is for ColumnTransformer
+                    msg = 'Length of tuple does not match assumptions'
+                    raise ValueError(msg)
+                if not isinstance(sub_component, (OpenMLFlow, type(None))):
+                    msg = 'Second item of tuple does not match assumptions. '\
+                          'Expected OpenMLFlow, got %s' % type(sub_component)
+                    raise ValueError(msg)
```

Comment:
Could you please make this a `TypeError`?

Reply:
Sure